Xamarin.Essentials/Contacts/Contacts.android.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Provider;
+using Net = Android.Net;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contacts
+    {
+        static async Task<Contact> PlatformPickContactAsync()
+        {
+            using var intent = new Intent(Intent.ActionPick);
+            intent.SetType(ContactsContract.CommonDataKinds.Phone.ContentType);
+            var result = await IntermediateActivity.StartAsync(intent, 101).ConfigureAwait(false);
+
+            if (result?.Data != null)
+                return PlatformGetContacts(result.Data);
+
+            return null;
+        }
+
+        internal static Contact PlatformGetContacts(Net.Uri contactUri)
+        {
+            if (contactUri == null)
+                return default;
+
+            using var context = Platform.GetCurrentActivity(true).ContentResolver;
+
+            using var cur = context.Query(contactUri, null, null, null, null);
+            var emails = new List<ContactEmail>();
+            var phones = new List<ContactPhone>();
+            var bDate = string.Empty;
+
+            if (cur.MoveToFirst())
+            {
+                var typeOfContact = cur.GetString(cur.GetColumnIndex(ContactsContract.CommonDataKinds.Phone.InterfaceConsts.Type));
+
+                var name = cur.GetString(cur.GetColumnIndex(ContactsContract.Contacts.InterfaceConsts.DisplayName));
+                string id;
+
+                id = cur.GetString(cur.GetColumnIndex(ContactsContract.CommonDataKinds.Email.InterfaceConsts.ContactId));
+                var idQ = new string[1] { id };
+
+                var projection = new string[2]
+                {
+                    ContactsContract.CommonDataKinds.Phone.Number,
+                    ContactsContract.CommonDataKinds.Phone.InterfaceConsts.Type,
+                };
+
+                var cursor = context.Query(
+                   ContactsContract.CommonDataKinds.Phone.ContentUri,
+                   null,
+                   ContactsContract.CommonDataKinds.Phone.InterfaceConsts.ContactId + "=?",
+                   idQ,
+                   null);
+
+                if (cursor.MoveToFirst())
+                {
+                    do
+                    {
+                        var phone = cursor.GetString(cursor.GetColumnIndex(projection[0]));
+                        var phoneType = cursor.GetString(cursor.GetColumnIndex(projection[1]));
+
+                        var contactType = GetPhoneContactType(phoneType);
+
+                        phones.Add(new ContactPhone(phone, contactType));
+                    }
+                    while (cursor.MoveToNext());
+                }
+                cursor.Close();
```

Comment:
In terms of code follow-ability, it is fine, but maybe this and the emails chunks can be moved to a static local method of say GetPhones() so that code reads a bit better?

Reply:
@pictos implemented this.